### PR TITLE
Add Story.Context

### PR DESCRIPTION
**Pull request by renatomassaro**
Story.Context is used for medium- to long-term persistence of arbitrary Storyline data.

Not to confuse with Story.Manager, which is used for structured data used throughout the entire Storyline. Short-lived data that span only one or a few steps should be persisted on Story.Step meta.

- [x] Automatic (transparent) handling of Helix IDs

---
This change is [Reviewable](https://reviewable.io/reviews/hackerexperience/helix/363)

### CODE REVIEW

File: lib/story/action/context.ex [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  `%{foo: %{bar: %{baz: 10}}}`
+  """
+  def save(entity = %Entity{}, field, subfield, value),
+    do: save(entity.entity_id, field, subfield, value)
+  def save(entity_id, field, subfield, id = %_{id: _}),
+    do: save(entity_id, field, subfield, Story.Context.store_id(id))
+  def save(entity_id, field, subfield, value) when not is_list(subfield),
+    do: save(entity_id, field, [subfield], value)
+  def save(entity_id, field, subfields, value) do
+    path = [field] ++ subfields
```
Comment (mememori) [inline lib/story/action/context.ex]: `path = [field| subfields]`

Comment (renatomassaro) [reply lib/story/action/context.ex]: :top:

**Comment by renatomassaro** [comment]
Reviewed 8 of 11 files at r1, 8 of 10 files at r2, 5 of 5 files at r3, 1 of 1 files at r4.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/363)*

**Comment by renatomassaro** [comment]
Reviewed 1 of 1 files at r5.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/hackerexperience/helix/363)*

**Pull request merged**